library/system/src/lib/yast2/system_service.rb

```diff
@@ -128,19 +128,24 @@ def start_mode=(mode)
     # The given value will be applied after calling #save.
     #
     # @param value [Boolean] true to set this service as active
-    def active=(active)
-      if active == service.active?
-        changes.delete(:active)
+    def active=(value)
+      if value == service.active?
+        unregister_change(:start_mode)
       else
-        changes[:active] = active
+        register_change(:active, value)
       end
     end
 
+    # Determine whether the service will be active after calling #save
+    #
+    # @return [Boolean] true if the service must be active; false otherwise
     def active
-      return changes[:active] unless changes[:active].nil?
+      return new_value_for(:active) if changed_value?(:active)
       service.active?
     end
 
+    alias_method :active?, :active
```

Comment:
Why not `active?` directly?

Reply:
It is a `leftover` (I kept them for compatibility reasons). But not needed anymore, so I've removed it.